pkg/sign/witness.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sign
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/sha256"
+	"io"
+
+	"github.com/digitorus/timestamp"
+	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
+	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+	tsaclient "github.com/sigstore/timestamp-authority/pkg/client"
+	tsagenclient "github.com/sigstore/timestamp-authority/pkg/generated/client/timestamp"
+)
+
+type Witness interface {
+	Witness(*protobundle.Bundle) error
+}
+
+type TimestampAuthority struct {
+	baseURL string
```

Comment:
We should include a context or common parameters like timeout to initialize the client.